### ISSUE TITLE
[extension API] devtools.panels docs clarity

### ISSUE THREAD

**Issue by paulirish** [GoogleChrome/devtools-docs#121]
https://developer.chrome.com/extensions/devtools_panels is a little hard to follow.

I'll try to summarize here. Two existing devtools panels can be extended: sources & elements:

``` js
chrome.devtools.panels.sources
chrome.devtools.panels.elements
```

The three APIs available to those objects are are:

``` js
chrome.devtools.panels.sources.onSelectionChanged.addListener
chrome.devtools.panels.sources.onSelectionChanged.removeListener
chrome.devtools.panels.sources.createSidebarPane
```

Then we also have these methods:

``` js
chrome.devtools.panels.setOpenResourceHandler
chrome.devtools.panels.openResource
chrome.devtools.panels.create
```

Then theres a slew of events available for panels and sidebar panes to bind to. And a few `set…` methods to display items. 

The docs page is pretty muddy at explaining this stuff so we could do a better job.

**Comment by nicolechung** [comment]
Yes! I would like to know if it's possible to open JS files from a devtools extension I am working on.

**Comment by spite** [comment]
What does "Fired when an object is selected in the panel" mean for the Sources panel? I can't get onSelectionChanged to fire no matter what I do in the Sources panel.

**Comment by Wizzy-wooz** [comment]
Hi guys,

can someone advise how can I call from my extension "Toggle Device Toolbar" panel like paulirish mentioned:

chrome.devtools.panels.sources
chrome.devtools.panels.elements

should it be chrome.devtools.panels.toggleDeviceToolbar?

Any help matters :)

Bet regards,

**Comment by trungdq88** [comment]
Same question: What does "Fired when an object is selected in the panel" mean for the Sources panel?

**Comment by paulirish** [comment]
`onSelectionChanged` should no longer be present on the `sourcesPanel`. It doesn't do anything. :)

No immediate plans to expand the events supported in sources panel, but contributions to chromium would be considered if you're up for that.

**Comment by tudorilisoi** [comment]
Imagine clicking on an error in devtools and have your favorite IDE open the (source-mapped) file and go to the faulty line. 
It could have been done with a small plugin for webpack(or any other dev server) and a little chrome devtools extension

**Comment by mervick** [comment]
Seems like `chrome.devtools.panels.setOpenResourceHandler` no longer available
